### PR TITLE
Grid building

### LINKED ISSUE #202: Implement grid building of the lv grid generator

### PR DESCRIPTION
**Pull request by t-ober**
Merge #184 and #219 first

Resolves #202

### CODE REVIEW

**Comment by t-ober** [comment]
There are some files here due to corrupt merge that are already in main but renamed. Reminder to self to check and delete the corresponding files from this PR

File: src/test/scala/edu/ie3/osmogrid/lv/region_coordinator/LvRegionCoordinatorIT.scala [hunk 1]
```diff
@@ -64,7 +65,9 @@ class LvRegionCoordinatorIT
           _.administrativeLevel shouldBe BoundaryAdminLevel.AMT_LEVEL
         )
         messages.foreach(_.lvConfig shouldBe lvConfig)
-        messages.foreach(_.replyTo shouldBe replyTo.ref)
+        messages.foreach(
+          _.lvCoordinatorRegionCoordinatorAdapter shouldBe regionCoordinatorReply.ref
+        )
```
Comment (julianhohmann) [inline src/test/scala/edu/ie3/osmogrid/lv/region_coordinator/LvRegionCoordinatorIT.scala:70]:
```suggestion
        messages.foreach(
          _.lvCoordinatorRegionCoordinatorAdapter shouldBe regionCoordinatorReply.ref
        )
        messages.foreach(
          _.lvCoordinatorGridGeneratorAdapter shouldBe gridGeneratorReply.ref
        )
```

File: src/test/scala/edu/ie3/osmogrid/lv/region_coordinator/LvRegionCoordinatorIT.scala [hunk 2]
```diff
@@ -118,7 +124,11 @@ class LvRegionCoordinatorIT
...
             administrativeLevel = adminLevel,
             lvConfig = lvConfigCapped,
-            replyTo = replyTo.ref
+            lvCoordinatorGridGeneratorAdapter =
+              lvCoordinatorGridGeneratorAdapter.ref,
+            lvCoordinatorRegionCoordinatorAdapter =
+              lvCoordinatorRegionCoordinatorAdapter.ref,
+            assetInformation = assetInformation
           )
         )
```
Comment (julianhohmann) [inline src/test/scala/edu/ie3/osmogrid/lv/region_coordinator/LvRegionCoordinatorIT.scala:134]: How can this test not fail? We are not using any MunicipalityCoordinator.

File: src/test/scala/edu/ie3/osmogrid/model/OsmoGridModelSpec.scala [hunk 3]
```diff
@@ -369,5 +370,38 @@ class OsmoGridModelSpec extends UnitSpec with OsmTestData {
...
+        OsmoGridModel.filterForClosedWays(input)
+
+      closedWays.size shouldBe 1
+      closedWays.headOption.getOrElse(
+        fail("Collection shouldn't be empty")
+      ) shouldBe ways.building1
+      closedWayNodes.size shouldBe 1
+      closedWayNodes.getOrElse(
+        nodes.building1Node1.id,
+        fail("Collection sholudn't be empty")
```
Comment (julianhohmann) [inline src/test/scala/edu/ie3/osmogrid/model/OsmoGridModelSpec.scala:402]:
```suggestion
        fail("Collection shouldn't be empty")
```

**Review by julianhohmann — CHANGES_REQUESTED** [review]
Great implementation of the LV grid generation. Thanks for your work! Just some minor issues and questions.

**Pull request merged**